### ISSUE TITLE
Added support of multiple scenarios in CModel validation

### ISSUE THREAD

**Issue by shidenko97** [yiisoft/yii#4335]
I know that Yii 1 not supported already, but I have great idea and hope you will consider it as a useful feature - to add support of multiple scenario of validation in CModel. It has backward compatibility and completed all old and new tests.

**Comment by samdark** [comment]
There are multiple considerations about it:

1. Yii 1.1 is in maintenance mode and significant enhancements aren't added anymore. @marcovtwout could consider it though...
2. Use case isn't clear. Usually it's a single scenario you use a model with. Would please elaborate it a bit. Maybe we're missing something great.
3. That's the same in Yii 2, only a single sceanario at once... so noone has the use case in 12 years since Yii was released.

**Comment by shidenko97** [comment]
> There are multiple considerations about it:
> 
> 1. Yii 1.1 is in maintenance mode and significant enhancements aren't added anymore. @marcovtwout could consider it though...
> 2. Use case isn't clear. Usually it's a single scenario you use a model with. Would please elaborate it a bit. Maybe we're missing something great.
> 3. That's the same in Yii 2, only a single sceanario at once... so noone has the use case in 12 years since Yii was released.

Well, so:

1. I've understood it and did it for myself at first, but then decided to offer it for all Yii1 community.
2. Personally me have project with long client's form, that has different validation scenarios depend from value of this forms. For example validating different type of documents (with different field in model for each type) and also validate kind of user's employment in this form. So I've faced with this problem and tried to solve. I can do it by another way - to create all combinations of scenario, but it will produce a*b*c*d*... rules, but solution in this PR produce only a+b+c+d... rules and is less complicated. Maybe creating custom validator for all scenarios and check needed condition in it will work to, but still not perfect (as for me). 
3. I haven't experience in Yii2, but I've researched the difference and it looks like `when` option will help in my situation, but unfortunately Yii1 hasn't it.
